src/plugins/discover/public/application/main/components/no_results/no_results_suggestions/no_results_suggestions.tsx

```diff
@@ -7,49 +7,129 @@
  */
 
 import React from 'react';
-import { EuiSpacer } from '@elastic/eui';
+import { EuiEmptyPrompt, EuiButton, EuiLoadingSpinner, EuiSpacer } from '@elastic/eui';
+import type { DataView } from '@kbn/data-views-plugin/common';
+import { isOfQueryType, isOfAggregateQueryType } from '@kbn/es-query';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useQuerySubscriber } from '@kbn/unified-field-list-plugin/public';
 import { NoResultsSuggestionDefault } from './no_results_suggestion_default';
 import {
   NoResultsSuggestionWhenFilters,
   NoResultsSuggestionWhenFiltersProps,
 } from './no_results_suggestion_when_filters';
 import { NoResultsSuggestionWhenQuery } from './no_results_suggestion_when_query';
 import { NoResultsSuggestionWhenTimeRange } from './no_results_suggestion_when_time_range';
+import { hasActiveFilter } from '../../layout/utils';
+import { useDiscoverServices } from '../../../../../hooks/use_discover_services';
+import { useFetchOccurrencesRange } from './use_fetch_occurances_range';
+import { NoResultsIllustration } from './assets/no_results_illustration';
 
 interface NoResultsSuggestionProps {
-  hasFilters?: boolean;
-  hasQuery?: boolean;
+  dataView: DataView;
   isTimeBased?: boolean;
   onDisableFilters: NoResultsSuggestionWhenFiltersProps['onDisableFilters'];
 }
 
-export function NoResultsSuggestions({
-  isTimeBased,
-  hasFilters,
-  hasQuery,
-  onDisableFilters,
-}: NoResultsSuggestionProps) {
-  const canAdjustSearchCriteria = isTimeBased || hasFilters || hasQuery;
+export const NoResultsSuggestions: React.FC<NoResultsSuggestionProps> = React.memo(
```

Comment:
To prevent unnecessary fetches

Reply:
I'm not sure I understand since I would have thought the `useEffect` around `fetchOccurrences` would take care of this issue, but wrapping it wouldn't cause a negative impact either way, so it should be fine to leave it there regardless.